packages/Box/Box.jsx

```diff
@@ -43,7 +43,7 @@ const inlineBetweenStyles = props =>
     { between: props.between, inline: props.inline },
     ({ between, inline }, breakpoint) => {
       const base = {
-        display: between ? 'flex' : 'block',
+        display: between || !inline ? 'flex' : 'block',
```

Comment:
Let's update this to `display: between !== undefined ? 'flex' : 'block'`

We want it to be `display: flex` whenever `between` is defined. Currently, when `between={0}`, `0` is coerced as a falsy value.